.circleci/config.yml

```diff
@@ -1504,6 +1504,120 @@ jobs:
+            sudo tar xzf cri-containerd-${CONTAINERD_VERSION}.linux-amd64.tar.gz -C /
+            sudo mkdir -p /etc/modules-load.d/
+            sudo mkdir -p /etc/sysctl.d/
+            sudo bash -c 'cat \<<EOF | sudo tee /etc/modules-load.d/k8s.conf
+            br_netfilter
+            EOF'
+            sudo bash -c 'cat \<<EOF | sudo tee /etc/sysctl.d/k8s.conf
+            net.bridge.bridge-nf-call-ip6tables = 1
+            net.bridge.bridge-nf-call-iptables = 1
+            EOF'
+            sudo sysctl --system
+      - run:
+          name: setup kubectl
+          command: |
+            curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/${K8S_VERSION}/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+            mkdir -p ${HOME}/.kube
+            touch ${HOME}/.kube/configcat
+            sudo mkdir -p  /etc/systemd/system/kubelet.service.d/
+            sudo bash -c 'cat \<<EOF > /etc/systemd/system/kubelet.service.d/0-containerd.conf
+            [Service]
+            Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --runtime-request-timeout=15m --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+            EOF'
+            sudo systemctl daemon-reload
+            sudo systemctl enable kubelet.service
+            sudo iptables -P FORWARD ACCEPT
+            sudo mkdir -p /etc/cni/net.d/
+      - run:
+          name: setup minikube
+          command: |
+            curl -Lo minikube https://github.com/kubernetes/minikube/releases/download/${MINIKUBE_VERSION}/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+      - run:
+          name: start minikube
+          command: |
+            sudo minikube start --driver=none --container-runtime=containerd --cpus 2 --memory 2048 --kubernetes-version=${K8S_VERSION} --extra-config="apiserver.service-account-key-file=/var/lib/minikube/certs/sa.pub" --extra-config="apiserver.service-account-signing-key-file=/var/lib/minikube/certs/sa.key"
+      - run:
+          name: wait for minikube
+          command: |
+            JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}';
+            until kubectl get nodes -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do
+              sleep 1;
+            done
+      - run:
+          name: fix RBAC
+          command: |
+            # make default account cluster-admin
+            kubectl create clusterrolebinding add-on-cluster-admin --clusterrole cluster-admin --serviceaccount=kube-system:default
+      - run:
+          name: dump cluster-info
+          command: |
+            kubectl cluster-info
+            kubectl get po --all-namespaces
+      - run:
+          name: Run Tests (with coverage)
+          command: |
+            source ./.circleci/smoketest_k8s.sh scalyr/scalyr-agent-ci-unittest:4 150 no_delete_existing_k8s_objs
```

Comment:
Hm, does our k8s verification script already verify that all the corresponding metrics are indeed retrieved and ingested?

If not, we should modify it to check for the presence of at least the basic metrics (ones which are present with Docker and CRI runtime).

Reply:
Our script does not check for metrics, and on top of that I can't build the image that runs the script anymore. I'm going to split off the smoketest work into its own PR/ticket because there is much more to it than I expected at this point.